From 08fff8e90dfc777109114a3a54e9fa79f106f064 Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Tue, 24 May 2016 17:29:48 +0800
Subject: [PATCH 1071/5242] MLK-12849 ARM: dts: imx6ull-ddr3-arm2: add a new
 dts file for TSC

commit  a624f107537ca3cb01b3aa255f8d98e725dfdd42 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since TSC has pin conflict with I2C1 which is used by PMIC and Camera,
we need to move TSC setting from LCDIF dts file into a separated one to
achieve the LCDIF and Camera feature in one DTS file. After the change,
we can get the supported features as follows.

-lcdif.dtb: lcd and camera, but no tsc
-tsc.dtb:   lcd and tsc, but no camera

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                         |    1 +
 .../arm/boot/dts/imx6ull-14x14-ddr3-arm2-lcdif.dts |   22 -----------
 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-tsc.dts  |   39 ++++++++++++++++++++
 3 files changed, 40 insertions(+), 22 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-tsc.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index de39a14..719e0fe 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -610,6 +610,7 @@ dtb-$(CONFIG_SOC_IMX6ULL) += \
 	imx6ull-14x14-ddr3-arm2-lcdif.dtb \
 	imx6ull-14x14-ddr3-arm2-ldo.dtb \
 	imx6ull-14x14-ddr3-arm2-qspi.dtb \
+	imx6ull-14x14-ddr3-arm2-tsc.dtb \
 	imx6ull-14x14-ddr3-arm2-gpmi-weim.dtb \
 	imx6ull-14x14-ddr3-arm2-uart2.dtb \
 	imx6ull-14x14-ddr3-arm2-usb.dtb \
diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-lcdif.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-lcdif.dts
index 1f12d12..40e86a4 100644
--- a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-lcdif.dts
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-lcdif.dts
@@ -27,25 +27,3 @@
 &pwm1 {
 	status = "okay";
 };
-
-&i2c1 {
-	status = "disabled";
-};
-
-&reg_usb_otg1_vbus {
-	pinctrl-0 = < >;
-	gpio = < >;
-};
-
-&usbotg1 {
-	status = "disabled";
-};
-
-&tsc {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_tsc>;
-	status = "okay";
-	xnur-gpio = <&gpio1 3 0>;
-	measure_delay_time = <0xfff>;
-	pre_charge_time = <0xffff>;
-};
diff --git a/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-tsc.dts b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-tsc.dts
new file mode 100644
index 0000000..8893e39
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ull-14x14-ddr3-arm2-tsc.dts
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6ull-14x14-ddr3-arm2-lcdif.dts"
+
+&csi {
+	status = "disabled";
+};
+
+&i2c1 {
+	status = "disabled";
+};
+
+&reg_usb_otg1_vbus {
+	pinctrl-0 = < >;
+	gpio = < >;
+};
+
+&ov5640 {
+	status = "disabled";
+};
+
+&usbotg1 {
+	status = "disabled";
+};
+
+&tsc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_tsc>;
+	status = "okay";
+	xnur-gpio = <&gpio1 3 0>;
+	measure_delay_time = <0xfff>;
+	pre_charge_time = <0xffff>;
+};
-- 
1.7.9.5

